e3/collection/dag.py

```diff
@@ -37,11 +37,21 @@ def __iter__(self):
         return self
 
     def next(self):
+        return self.next_element(with_predecessors=False)
+
+    def next_element(self, with_predecessors=False):
         """Retrieve next element in topological order.
 
+        :param with_predecessors: if True return also of the predecessors of
+            the returned node
+        :type with_predecessors: bool
         :return: a tuple id, data. (None, None) is returned if no element is
-            available
-        :rtype: (str, object)
+            available. If with_predecessors is set to True then an predecessors
+            are added as third element of the tuple and in that case if no
+            element
+            is available (None, None, None) is returned.
+        :rtype: (str, object) | (str, object, list[str]) |
```

Comment:
I think you should replace object with T (for generic type). See https://www.jetbrains.com/help/pycharm/2016.3/type-hinting-in-pycharm.html